algviz/interface/visitors.py

```diff
@@ -190,3 +190,29 @@ class WidgetVisitor(Visitor):
 
     def visit(self, *args, **kwargs):
         return super().visit(*args, **kwargs)
+
+class LinkedListVisitor(Visitor):
+    """Linked lists are represented as an array of an array and a current index, i.e.,
```

Comment:
I'd prefer if the `LinkedListVisitor` superclass didn't implement these methods, since there's no build-in linked-list in python.
A concrete subclass of `LinkedListVisitor` with this behavior would be lovely (modulo other feedback) but belongs in weird_visitors.py.